web/src/main/java/org/phoenixctms/ctsms/web/component/datatable/ColumnManagementBean.java

```diff
@@ -0,0 +1,233 @@
+			}
+		}
+		return true; //rowtoggler, buttoncolumn, dropdown column, ...
+	}
+
+	private void appendRequestContextCallbackArgs(boolean operationSuccess) {
+		RequestContext requestContext = RequestContext.getCurrentInstance();
+		if (requestContext != null) {
+			requestContext.addCallbackParam(JSValues.AJAX_OPERATION_SUCCESS.toString(), operationSuccess);
+		}
+	}
+
+	public void update(ActionEvent ae) {
+		String dataTableClientId = (String) ae.getComponent().getAttributes().get("datatable_id");
+		Iterator<Entry<String, String>> it = getVisibleMap(dataTableClientId).entrySet().iterator();
+		LinkedHashSet<DataTableColumnVO> in = new LinkedHashSet<DataTableColumnVO>();
+		int visibleCount = 0;
+		while (it.hasNext()) {
+			Entry<String, String> value = it.next();
+			DataTableColumnVO tableColumnVO = new DataTableColumnVO();
+			tableColumnVO.setTableName(dataTableClientId);
+			tableColumnVO.setColumnName(value.getKey());
+			tableColumnVO.setVisible(Boolean.parseBoolean(value.getValue()));
+			if (tableColumnVO.getVisible()) {
+				visibleCount++;
+			}
+			in.add(tableColumnVO);
+		}
+		if (visibleCount > 0) {
+			try {
+				updateVisibleMap(dataTableClientId,
+						WebUtil.getServiceLocator().getUserService().setDataTableColumns(WebUtil.getAuthentication(), WebUtil.getUser().getId(), in));
+				if (!FILTER_INVISIBLE) {
+					((DataTable) WebUtil.findComponentByClientId(dataTableClientId)).clearFilters();
+				}
+				Messages.addLocalizedMessage(FacesMessage.SEVERITY_INFO, MessageCodes.UPDATE_OPERATION_SUCCESSFUL);
+				appendRequestContextCallbackArgs(true);
+				return;
+			} catch (ServiceException e) {
+				Messages.addMessage(FacesMessage.SEVERITY_ERROR, e.getMessage());
+			} catch (AuthenticationException e) {
+				Messages.addMessage(FacesMessage.SEVERITY_ERROR, e.getMessage());
+				WebUtil.publishException(e);
+			} catch (AuthorisationException e) {
+				Messages.addMessage(FacesMessage.SEVERITY_ERROR, e.getMessage());
+			} catch (IllegalArgumentException e) {
+				Messages.addMessage(FacesMessage.SEVERITY_ERROR, e.getMessage());
+			}
+		} else {
+			Messages.addLocalizedMessage(FacesMessage.SEVERITY_ERROR, MessageCodes.NO_VISIBLE_COLUMNS_SELECTED);
+		}
+		reloadVisibleMap(dataTableClientId);
+		appendRequestContextCallbackArgs(false);
+	}
+
+	public void clear(ActionEvent ae) {
+		String dataTableClientId = (String) ae.getComponent().getAttributes().get("datatable_id");
+		try {
+			WebUtil.getServiceLocator().getUserService().clearDataTableColumns(WebUtil.getAuthentication(), WebUtil.getUser().getId(), dataTableClientId, null);
+			clearVisibleMap(dataTableClientId);
+			if (!FILTER_INVISIBLE) {
+				((DataTable) WebUtil.findComponentByClientId(dataTableClientId)).clearFilters();
+			}
+			Messages.addLocalizedMessage(FacesMessage.SEVERITY_INFO, MessageCodes.UPDATE_OPERATION_SUCCESSFUL);
+			appendRequestContextCallbackArgs(true);
+			return;
+		} catch (ServiceException e) {
```

Comment:
Java 7 offers https://docs.oracle.com/javase/8/docs/technotes/guides/language/catch-multiple.html 
```java
catch (IllegalArgumentException|AuthorisationException|ServiceException e) {...
```

Reply:
ok that is in fact a helpful syntax, fasible and applicable to all occurrences. i suggest to do a batch refactor for this with a subsequent commit.

Reply:
https://github.com/phoenixctms/ctsms/issues/52